### DISCUSSION TITLE
enabling undercurl in neovim for windows

### DISCUSSION THREAD

**Discussion by ajmandourah** [microsoft/terminal#16696] [Q&A]
So udnercurl support has been added to the preview release. I am currently using neovim for windows, but in neovim for windows the undercurl is sill not shown, instead all i see is a white coloried underlines. 
i dunno if this is related but it seem like windows is using ` as an escape char where unix system uses \. i tried configureing neovim with:

```
    vim.api.nvim_set_var('t_Cs', '\\e[4:3m')
    vim.api.nvim_set_var('t_Ce', '\\e[4:0m')
```

and 

```
    vim.api.nvim_set_var('t_Cs', '`e[4:3m')
    vim.api.nvim_set_var('t_Ce', '`e[4:0m')
```

with no luck 

any ideas? this is mainly for windows as wsl and tmux have their own config and should work.

**Comment by Kayzels** [comment]
There's a pending pull request in Neovim that enables this. From the above, it looks like you're running Neovim in Powershell? Because the escape sequences differ per shell.

Check out neovim/neovim#27629

Edit: Oops, I referred to your own pull request! I'm keeping this comment, though, for those (like me!) who see this discussion, but not the resolution.

**Reply by DHowett** [reply]
> Because the escape sequences differ per shell.

They absolutely should not do that. The shell has very little control over what escape sequences are accepted by the terminal emulator. All software running under Windows Terminal support undercurl as of Terminal 1.19; all software running in the vintage Windows Console support undercurl as of the (as yet unreleased) Windows 11 2024 Update

**Reply by tusharsnx** [reply]
I think they heard you: https://github.com/neovim/neovim/pull/28052
